scripts/profiler.sh

```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
```

Comment:
Profiler reports are produced by the CI and added as build artifacts which can be downloaded (see the bottom of the page https://github.com/ChainSafe/mina-rs/actions/runs/1579095502)

A benefit of committing these to the repo would be to keep a log of them over time which I am not sure if github will guarantee. @hanabi1224 any thoughts?

Reply:
👍🏼

Reply:
Are you able set an alert on these reports if a commit causes a significant decrease in performance?

Reply:
I can cache the benchmark result of the previous run so that the subsequent reports will be generated in comparison mode.  ATM, I don't think either criterion crate or iai crate provides the functionality of alerting, will investigate more

Reply:
Also this might not be feasible because GH actions use azure [DSv2-series](https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series) VMs which has a pool of CPU models released from 2013-2019, it's not possible to fix a CI job to a certain CPU so that either the comparison or the alert tend to be invalid